src/utils/web.py

```diff
@@ -145,7 +147,12 @@ def getUrlFd(url, headers=None, data=None, timeout=None):
             request.add_data(data)
         httpProxy = force(proxy)
         if httpProxy:
-            request.set_proxy(httpProxy, 'http')
+            proxyHandler = ProxyHandler({
+                'http': httpProxy,
+                'https': httpProxy
+                })
+            proxyOpener = build_opener(proxyHandler)
+            install_opener(proxyOpener)
```

Comment:
> Correct, which is the same way it presently works without this commit.
> Any plugin that does not currently make use of util-web for HTTP(S) requests will not benefit from this commit.

I was thinking about changing the proxy config.

Oh and by the way, I just noticed your code makes it impossible for users to disable the proxy.

> Just to confirm, is your main concern simply that the activation of a global HTTP(S) proxy should rather be done in within some configuration-related logic so that global proxy changes reflect immediately?

Yes. You can do that by adding a config type (ie. class) and overriding `set` or `setValue` (I never remember which one)